modules/server/src/main/scala/scaladex/server/route/ProjectPages.scala

```diff
@@ -111,6 +111,43 @@ class ProjectPages(env: Env, database: WebDatabase, searchEngine: SearchEngine)(
           }
         }
       },
+
+      get {
+        path(projectM / "artifacts" ) { ref =>
+          artifactsParams { params =>  
+            getProjectOrRedirect(ref,user) { project => 
+              val headerF = service.getProjectHeader(project)
+              for {
+                header <- headerF
+              } yield {
+                val allArtifacts = header.toSeq.flatMap(_.artifacts)
+
+                // Group by artifact name
+                val artifactsByName = allArtifacts.groupBy(_.artifactName)
+
+                val groupedArtifacts = artifactsByName
+                  .groupBy { case (artifactName, artifacts) =>  
+                    artifacts.maxBy(_.version).version 
+                  }
+                  .map { case (latestVersion, artifactsByName) =>
+                    latestVersion -> artifactsByName
+                      .map { case (name, artifacts) => name -> artifacts
+                        .filter(_.version == latestVersion)
+                      }
+                      .filter { case(_, artifacts) =>
+                        params.binaryVersions.forall( binaryVersion => artifacts
+                          .exists(_.binaryVersion == binaryVersion))
+                      }
+                }
```

Comment:
After this `map` you should `filter` if an `artifactsByName` is empty. Then you will be able to remove the `@if(!artifactsByName.isEmpty){` in the HTML view.